lib/register_client.rb

```diff
@@ -109,6 +111,24 @@ def get_records_with_history_for_entry_type(since_entry_number, entry_type)
       records_with_history
     end
 
+    def get_register_proof(register, phase)
+      @register_proof = JSON.parse(RestClient.get("https://#{register}.#{phase}.openregister.org/proof/register/merkle:sha-256"))
+    end
+
+    def validate_register_integrity(rsf, current_entry_number)
+      rsf_lines = rsf.split("\n")
+      latest_root_hash = rsf_lines[rsf_lines.length - 1].split("\t")[1]
+      register_proof = get_register_proof(@register, @phase)
+
+      if (register_proof['total-entries'] < current_entry_number)
```

Comment:
Do we not want a `!=` comparison here too? i.e. any different value is invalid?

Reply:
No, because it may be the case that the register proof has new entries, which means that it will be greater than the current entry number. In this case, we proceed on to the next check which compares the hash from the register proof against the hash at the end of the downloaded RSF, and if they're not equal, throw an error, otherwise ingest that latest RSF.